zend_abstract_interface/config/config_decode.c

```diff
@@ -117,7 +117,20 @@ static bool zai_config_is_valid_int_format(const char *str) {
 static bool zai_config_decode_int(zai_string_view value, zval *decoded_value) {
     if (!zai_config_is_valid_int_format(value.ptr)) return false;
 
-    ZVAL_LONG(decoded_value, zend_atol(value.ptr, value.len));
+#if PHP_VERSION_ID >= 80200
+    zend_string *zstr = zend_string_init(value.ptr, value.len, 0);
+    zend_string *err = NULL;
+    zend_long l = zend_ini_parse_quantity(zstr, &err);
+    if (err) {
+        // what to do with err?
+        zend_string_free(err);
+    }
```

Comment:
Given that we anyway check zai_config_is_valid_int_format() to be integers with a possibly leading - only (incl. spaces), this basically should never fail. So, it should not be too important, just free and `return false;` in that case I'd say.

Reply:
Thanks, I've updated the PR and left a comment about this.